apps/sms/views/conversation/js/conversation.js

```diff
@@ -1203,13 +1211,15 @@ var ConversationView = {
 
     // We're waiting for the keyboard to disappear before animating back
     return this._ensureKeyboardIsHidden().then(function() {
+      var inConversation = Navigation.isCurrentPanel('thread');
       // Need to assimilate recipients in order to check if any entered
       this.assimilateRecipients();
 
       // If we're leaving a thread's message view,
       // ensure that the thread object's unreadCount
       // value is current (set = 0)
-      if (Threads.active) {
+      if (inConversation) {
+        // TODO: Will need to replace with Conversation service method.
         Threads.active.unreadCount = 0;
```

Comment:
you missed my comment here :) can we replace `Threads.active` by `this.activeThread` here ?